plugin.py

```diff
@@ -404,18 +404,29 @@ def run(self):
 
     def selected_unit_test_method_names(self, view):
         """
-        If all selections are test methods returns an array of all selected
-        test method names; otherwise None
+        Returns an array of selected test method names.
+        Selection can be anywhere inside one or more test methods.
+        If no selection is found inside any test method, then all test method names are returned.
         """
 
-        # @todo should be a scoped selection i.e. is the selection a source.php entity.name.function
         method_names = []
-        for region in view.sel():
-            word = view.substr(view.word(region))
-            if not is_valid_php_identifier(word) or word[:4] != 'test':
-                return None
+        function_areas = view.find_by_selector('meta.function')
+        function_regions = view.find_by_selector('entity.name.function')
 
-            method_names.append(word)
+        for region in view.sel():
+            for i, area in enumerate(function_areas):
+                if not area.a <= region.a <= area.b:
+                    continue
+                word = view.substr(function_regions[i])
+                if is_valid_php_identifier(word):
+                    method_names.append(word)
+                break
+
+        if not method_names:
```

Comment:
No, each file extends a singe TestCase base class; each test case includes multiple tests. It's an early project so there are only a few small tests.

This feature was primary made to resolve #49 so that the plugin would work with _annotated_ tests. By being able to select anywhere inside a test method to run it, rather than specifically the method name, the naming convention of the test method no longer matters for the package to work properly. This is the primary change of this feature. The package works for the developer rather than making the developer modify their coding method to suit the package.

![image](https://cloud.githubusercontent.com/assets/134939/23021967/01c1a7d4-f425-11e6-94e9-a8c110e6cf41.png)

The `Run all Tests` command runs _all_ test case files in the project.
The `Run Single Test` runs either the selected test/tests in a test case, or the entire single test case class if the selection is invalid or none.

You are right the new functionality is the same as the old, but it is less restrictive as it tries to suit the _intention_ of the developer. It's much quicker to select anywhere inside of a unit-test method to run it, rather than only selecting inside the method name, especially if selecting more than one test to run.